packages/test-base/src/androidTest/kotlin/io/realm/kotlin/test/shared/RealmInMemoryTests.kt

```diff
@@ -0,0 +1,243 @@
+        val tmpDir2 = PlatformUtils.createTempDir("2")
+        val configuration2 = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .directory(tmpDir2)
+            .inMemory()
+            .build()
+        val realm2 = Realm.open(configuration2)
+        realm2.writeBlocking {
+            copyToRealm(Sample().apply { stringField = "bar" })
+        }
+        assertEquals(1, realm.query(Sample::class).count().find())
+        assertEquals("foo", realm.query<Sample>("stringField == 'foo'").find().first().stringField)
+        assertEquals(1, realm2.query(Sample::class).count().find())
+        assertEquals("bar", realm2.query<Sample>("stringField == 'bar'").find().first().stringField)
+        realm2.close()
+        PlatformUtils.deleteTempDir(tmpDir2)
+    }
+
+    @Test
+    fun inMemoryRealm_delete() {
+        try {
+            Realm.deleteRealm(realm.configuration)
+            fail("Realm.deleteRealm should fail with illegal state")
+        } catch (ignored: IllegalStateException) {
+        }
+
+        // Nothing should happen when deleting a closed in-mem-realm.
+        realm.close()
+        Realm.deleteRealm(realm.configuration)
+    }
+
+    @Test
+    fun inMemoryRealm_writeCopyTo() {
+        val key: ByteArray = TestHelper.getRandomKey()
+        val fileName: String = tmpDir + ".realm"
+        val encFileName: String = tmpDir + ".enc.realm"
+
+        val conf = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .directory(fileName)
+            .build()
+        val encConf = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .directory(encFileName)
+            .encryptionKey(key)
+            .build()
+        Realm.deleteRealm(conf)
+        Realm.deleteRealm(encConf)
+        realm.writeBlocking {
+            copyToRealm(Sample().apply { stringField = "foo" })
+        }
+
+        // Tests a normal Realm file.
+        realm.writeCopyTo(conf)
+        val onDiskRealm: Realm = Realm.open(conf)
+        assertEquals(1, onDiskRealm.query<Sample>().count().find())
+        onDiskRealm.close()
+
+        // Tests a encrypted Realm file.
+        realm.writeCopyTo(encConf)
+        val onDiskEncryptedRealm: Realm = Realm.open(encConf)
+        assertEquals(1, onDiskEncryptedRealm.query<Sample>().count().find())
+        onDiskEncryptedRealm.close()
+
+        // Tests with a wrong key to see if it fails as expected.
+        val randomKey = Random.nextBytes(64)
+        RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .directory(encFileName)
+            .encryptionKey(randomKey)
+            .build()
+            .let { conf ->
+                assertFailsWith(IllegalArgumentException::class, "Encrypted Realm should not be openable with a wrong encryption key") {
+                    Realm.open(conf)
+                }
+            }
+    }
+
+    // Tests writeCopyTo result when called in a transaction.
```

Comment:
This behaves somewhat differently from the Java SDK, my suspicion is that the semantics differ between

`realm.beginTransaction()` and `realm.commitTransaction()`

in comparison to `realm.writeBlocking{}`

Reply:
the key difference is that the assertEquals in the test has a value of 1 in realm Java while in kotlin it is 0

Reply:
Yes, If I remember correctly there is a difference between Kotlin and Java here, so this is fine.